qiling/os/windows/dlls/kernel32/fileapi.py

```diff
@@ -392,10 +394,11 @@ def hook_CreateDirectoryA(ql, address, params):
     try:
         target_dir = os.path.join(ql.rootfs, lpPathName.replace("\\", os.sep))
         print('TARGET_DIR = %s' % target_dir)
-        ql.os.transform_to_real_path(lpPathName) 
+        ql.os.transform_to_real_path(lpPathName)
         # Verify the directory is in ql.rootfs to ensure no path traversal has taken place
-        if os.path.commonprefix(target_dir, ql.rootfs) != ql.rootfs:
-            ql.dprint(D_INFO, 'CreateDirectoryA attempted to create dir outside of rootfs: "%s". Not creating.' % (target_dir))
+        if os.path.commonprefix([target_dir, ql.rootfs]) != ql.rootfs:
+            ql.dprint(D_INFO,
+                      'CreateDirectoryA attempted to create dir outside of rootfs: "%s". Not creating.' % target_dir)
```

Comment:
this technically never exits. we dont have to print. rootfs is the "OS" it should not escape anyway.

Reply:
15db38b